spec/requests/api/v2/people_controller_index_spec.rb

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::V2::PeopleController do
+  let(:supplier) { create(:supplier) }
+  let!(:application) { create(:application, owner_id: supplier.id) }
+  let!(:access_token) { create(:access_token, application: application).token }
+  let(:headers) { { 'CONTENT_TYPE': content_type }.merge('Authorization' => "Bearer #{access_token}") }
+  let(:response_json) { JSON.parse(response.body) }
+  let(:content_type) { ApiController::CONTENT_TYPE }
+
+  describe 'GET /people' do
+    let(:schema) { load_yaml_schema('get_people_responses.yaml', version: 'v2') }
+    let!(:people) { create_list :person, 2 }
+    let(:params) { {} }
+
+    context 'when successful' do
+      context 'when there are no params' do
+        before { get '/api/v2/people', params: params, headers: headers }
+
+        it_behaves_like 'an endpoint that responds with success 200'
+
+        it 'returns correct attributes' do
+          expect(response_json['data'].first['attributes']).to include(
+            'first_names',
+            'last_name',
+            'date_of_birth',
+            'gender_additional_information',
+            'nomis_prison_number',
+            'prison_number',
+            'criminal_records_office',
+            'police_national_computer',
+          )
+        end
+
+        it 'returns the correct number of people' do
+          expect(response_json['data'].count).to eq(2)
+        end
+      end
+
+      describe 'filtering results by police_national_computer' do
+        let!(:person) { create(:person, police_national_computer: 'AB/1234567') }
+        let(:filters) do
+          {
+            bar: 'bar',
+            police_national_computer: 'AB/1234567',
+            foo: 'foo',
+          }
+        end
+        let(:params) { { filter: filters } }
+
+        before { get '/api/v2/people', params: params, headers: headers }
+
+        it 'returns the correct number of people' do
+          expect(response_json['data'].size).to eq(1)
+        end
+
+        it 'returns the person that matches the filter' do
+          expect(response_json).to include_json(data: [{ id: person.id }])
+        end
+      end
+
+      describe 'filtering results by multiple filters' do
```

Comment:
This definitely doesn't cover the possible options for the identifier filters. I'm wondering whether we need to do these tests here and whether we'd benefit from removing most of these specs and validating handling of outbound messages to the finder?

Reply:
I think it's fine that it does not cover all the possible options, the unit test `spec/services/v2/people/finder_spec.rb` covers all the cases.
but I agree with you, it feels we are testing this again and again in any endpoints, they still have some values, we could spy that those filters are called (for faster tests), of we could use the shared example, something like `it behaves like an endpoint that supports filters`, 
this is something we should do, maybe not just now.